newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+import axios from 'axios';
+import {
+  GDevelopProjectApi,
+  GDevelopProjectResourcesStorage,
+} from './ApiConfigs';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+
+export const CLOUD_PROJECT_NAME_MAX_LENGTH = 50;
+
+const projectResourcesClient = axios.create({
+  withCredentials: true,
+});
+
+type CloudProject = {|
+  id: string,
+  name: string,
+  createdAt: string,
+  currentVersion?: string,
+  deletedAt?: string,
+|};
+
+export type CloudProjectWithUserAccessInfo = {|
+  ...CloudProject,
+  lastModifiedAt: string,
+|};
+
+const refetchCredentialsForProjectAndRetryIfFailed = async <T>(
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string,
+  apiCall: () => Promise<T>
+): Promise<T> => {
+  try {
+    const response = await apiCall();
+    return response;
+  } catch (error) {
+    if (error.response && error.response.status === 403) {
+      await getCredentialsForCloudProject(authenticatedUser, cloudProjectId);
+      const response = await apiCall();
+      return response;
+    }
+    throw error;
+  }
+};
+
+const getVersionIdFromPath = (path: string): string => {
```

Comment:
you call that versionId, and yet you're retrieving the filename. Is it the same thing?

Reply:
Well the version is the filename of the zip file to it kinda makes sense

Reply:
ok, I got confused to see different things named the same at first read. If you know it it's fine though.
Maybe it's worth a tiny comment for future contributors? // the filename is in the format `xxxid.zip`?

Reply:
Good idea!